File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/search/Acl.java
package com.algolia.model.search;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Gets or Sets acl */
@JsonAdapter(Acl.Adapter.class)
public enum Acl {
  ADD_OBJECT("addObject"),

  ANALYTICS("analytics"),

  BROWSE("browse"),

  DELETE_OBJECT("deleteObject"),

  DELETE_INDEX("deleteIndex"),

  EDIT_SETTINGS("editSettings"),

  LIST_INDEXES("listIndexes"),

  LOGS("logs"),

  PERSONALIZATION("personalization"),

  RECOMMENDATION("recommendation"),

  SEARCH("search"),

  SEE_UNRETRIEVABLE_ATTRIBUTES("seeUnretrievableAttributes"),

  SETTINGS("settings"),

  USAGE("usage");

  private final String value;

  Acl(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static Acl fromValue(String value) {
    for (Acl b : Acl.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<Acl> {

    @Override
    public void write(final JsonWriter jsonWriter, final Acl enumeration)
      throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public Acl read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return Acl.fromValue(value);
    }
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/Acl.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * Acl Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class Acl
{
    /**
     * Possible values of this enum
     */
    const ADD_OBJECT = 'addObject';

    const ANALYTICS = 'analytics';

    const BROWSE = 'browse';

    const DELETE_OBJECT = 'deleteObject';

    const DELETE_INDEX = 'deleteIndex';

    const EDIT_SETTINGS = 'editSettings';

    const LIST_INDEXES = 'listIndexes';

    const LOGS = 'logs';

    const PERSONALIZATION = 'personalization';

    const RECOMMENDATION = 'recommendation';

    const SEARCH = 'search';

    const SEE_UNRETRIEVABLE_ATTRIBUTES = 'seeUnretrievableAttributes';

    const SETTINGS = 'settings';

    const USAGE = 'usage';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::ADD_OBJECT,
            self::ANALYTICS,
            self::BROWSE,
            self::DELETE_OBJECT,
            self::DELETE_INDEX,
            self::EDIT_SETTINGS,
            self::LIST_INDEXES,
            self::LOGS,
            self::PERSONALIZATION,
            self::RECOMMENDATION,
            self::SEARCH,
            self::SEE_UNRETRIEVABLE_ATTRIBUTES,
            self::SETTINGS,
            self::USAGE,
        ];
    }
}

